app/src/main/kotlin/compiler/semantic_analysis/VariablePropertiesAnalyzer.kt

```diff
@@ -111,6 +115,12 @@ object VariablePropertiesAnalyzer {
                         }
                     }
                     node.body.forEach { analyzeVariables(it, node) }
+                    if (node.returnType != Type.Unit)
+                        mutableVariableProperties[functionReturnedValueVariables[node]!!] = MutableVariableProperties(
+                            node,
+                            referenceHashSetOf(node),
```

Comment:
Are you sure about `accessedIn`? Shouldn't this be some caller functions?

Reply:
I think so, this variable is created to store the result of a given function, and in the epilogue its value is moved to `rax` from this variable's location, so it shouldn't be possible to access directly this variable from other scopes. Added comment in the code for clarity.

Reply:
Great, thanks